src/gui/widgets/VectorGraph.cpp

```diff
@@ -0,0 +1,777 @@
+										model()->getLastModifiedTensionType()
+									);
+		model()->insertPointAfter(leftBoundIndex, newPoint);
+		model()->setCurrentDraggedPoint(leftBoundIndex + 1);
+		event->accept(); // maybe unnecessary
+		update();
+	}
+	else if (event->button() == Qt::MouseButton::LeftButton)
+	{
+		int pointIndex = model()->getPointIndexFromCoords(coordToRawX(event->x()) * m_width, coordToRawY(event->y()) * m_height, m_width, m_height);
+		int tensionHandleIndex = model()->getPointIndexFromTensionHandleCoords(coordToRawX(event->x()) * m_width, coordToRawY(event->y()) * m_height, m_width, m_height);
+
+		if (pointIndex > -1)
+		{
+			model()->setCurrentDraggedPoint(pointIndex);
+		}
+		else if (tensionHandleIndex > -1)
+		{
+			setCursor(Qt::BlankCursor);
+			model()->setStoredCursorPos(cursor().pos());
+			model()->setCurrentDraggedTensionHandle(tensionHandleIndex);
+		}
+	}
+}
+
+void VectorGraph::mouseMoveEvent(QMouseEvent *event)
+{
+	if (event->modifiers() == Qt::KeyboardModifier::AltModifier)
+		model()->setIsGridSnapEnabled(true);
+	else
+		model()->setIsGridSnapEnabled(false);
+
+	if (model()->getCurrentDraggedPoint() != -1)
+	{
+		model()->tryMove(model()->getCurrentDraggedPoint(), coordToRawX(event->x()), coordToRawY(event->y()));
+		update();
+	}
+
+	if (model()->getCurrentDraggedTensionHandle() != -1)
+	{
+		QCursor c = cursor();
+		float delta = c.pos().y() - model()->getStoredCursorPos().y();
+		c.setPos(model()->getStoredCursorPos());
+		setCursor(c);
+
+		int index = model()->getCurrentDraggedTensionHandle();
+		VectorGraphPoint * previousPoint = model()->getPoint(index - 1);
+		VectorGraphPoint * pointToEdit = model()->getPoint(index);
+
+		if (previousPoint->y() > pointToEdit->y())
+			delta *= -1;
+
+		// Subtracting, moving down vertically makes the y value go up
+		float newTension = pointToEdit->tension() - delta / 250; // Make adjustable from somewhere else - this is an important tweak
+		if (newTension > 1)
+			newTension = 1;
+		else if (newTension < -1)
+			newTension = -1;
+		pointToEdit->setTension(newTension);
+		update();
+	}
+}
+
+void VectorGraph::mouseReleaseEvent(QMouseEvent * event)
+{
+	if (model()->getCurrentDraggedPoint() != -1)
+		model()->resetCurrentDraggedPoint();
+
+	if (model()->getCurrentDraggedTensionHandle() > -1)
+	{
+		VectorGraphPoint * point = model()->getPoint(model()->getCurrentDraggedTensionHandle());
+		model()->setLastModifiedTension(point->tension());
+		model()->setLastModifiedTensionType(point->tensionType());
+
+		QCursor c = cursor();
+		QPoint newCursorPoint(
+					rawToCoordX(getTensionHandleXVal(model()->getCurrentDraggedTensionHandle())),
+					rawToCoordY(getTensionHandleYVal(model()->getCurrentDraggedTensionHandle())));
+		c.setPos(mapToGlobal(newCursorPoint));
+		c.setShape(Qt::ArrowCursor);
+		setCursor(c);
+		model()->resetCurrentDraggedTensionHandle();
+	}
+}
+
+bool VectorGraph::eventFilter(QObject *watched, QEvent *event)
+{
+	if (event->type() == QEvent::ContextMenu)
```

Comment:
Instead of using `eventFilter`, you can override [`contextMenuEvent`](http://doc.qt.io/qt-5/qwidget.html#contextMenuEvent).